app/models/report.rb

```diff
@@ -14,6 +16,18 @@ class Report < ActiveRecord::Base
 
   after_commit :send_report_notification, :on => :create
 
+  def item
+    if item_type == POST
+      Post.find_by(id: item_id)
+    elsif item_type == COMMENT
+      Comment.find_by(id: item_id)
+    end
+  end
```

Comment:
Rails will do the magic for you if `item_type` contains the right thing: http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#module-ActiveRecord::Associations::ClassMethods-label-Polymorphic+Associations